github/github.go

```diff
@@ -64,11 +65,8 @@ type Client struct {
 	// User agent used when communicating with the GitHub API.
 	UserAgent string
 
-	// Rate specifies the current rate limit for the client as determined by the
-	// most recent API call.  If the client is used in a multi-user application,
-	// this rate may not always be up-to-date.  Call RateLimits() to check the
-	// current rate.
-	Rate Rate
+	rateMu sync.Mutex
```

Comment:
Yes, I chose `sync.Mutex` over `sync.RWMutex` here on purpose, because I estimated that it'd be simpler and faster, and hence more appropriate.

I've seen some arguments and articles demonstrating that the more complex read-write locks are often slower than the simpler mutexes for basic use cases:

- https://blogs.oracle.com/roch/entry/beware_of_the_performance_of
- http://preshing.com/20111124/always-use-a-lightweight-mutex/
- http://askldjd.com/2011/03/06/performance-comparison-on-reader-writer-locks/

I was convinced to at least use `sync.Mutex` by default, and upgrade to `sync.RWMutex` after doing profiling and ensuring it's actually faster and better.

If you prefer I change this to `sync.RWMutex`, I can do that.